minion/server/core.py

```diff
@@ -172,3 +173,15 @@ def stop_server():
     if func is None:
         raise RuntimeError('Not running with the Werkzeug Server')
     func()
+
+def ensure_mongo(func):
```

Comment:
Okay, this won't work yet.
First, flask needs your decorator wrapper function be be wrapped correctly by using `functools.wraps()`. 

In this case this would like something like:

```python
from functools import wraps

...

def ensure_mono(func):
    @wraps(func)
    def wrapper(*args, **kwargs): 
        ...
```

In every case it's good practice to use `functools.wraps()` - It makes your wrapper appear like the wrapped function. Which is what you want in most of the cases.

Next, you forgot to call the wrapped route function in case there actually is a mongodb server running. I suggest to use the `else` block of your `try/except` statement. It could look something like this:

```python
try:
    ...
except ....:
    ...
else:
    return jsonify({'data': 'No mongodb ...'}), 500
finally:
    ...
```

The `return` statement in the `else`-block will be used a flask Response.